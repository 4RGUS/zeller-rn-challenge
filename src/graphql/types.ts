export interface TableStringFilterInput {
  ne?: string;
  eq?: string;
  le?: string;
  lt?: string;
  ge?: string;
  gt?: string;
  contains?: string;
  notContains?: string;
  between?: string[];
  beginsWith?: string;
}

export interface TableZellerCustomerFilterInput {
  id?: TableStringFilterInput;
  name?: TableStringFilterInput;
  email?: TableStringFilterInput;
  role?: TableStringFilterInput;
}

export interface ZellerCustomer {
  id: string;
  name?: string;
  email?: string;
  role?: string;
}

export interface ZellerCustomerConnection {
  items?: ZellerCustomer[];
  nextToken?: string;
}

export interface ListZellerCustomersQueryVariables {
  filter?: TableZellerCustomerFilterInput;
  limit?: number;
  nextToken?: string;
}

export interface ListZellerCustomersQuery {
  listZellerCustomers?: ZellerCustomerConnection;
}

export interface GetZellerCustomerQueryVariables {
  id: string;
}

export interface GetZellerCustomerQuery {
  getZellerCustomer?: ZellerCustomer;
}

export enum Roles {
  ADMIN = "ADMIN",
  MANAGER = "MANAGER",
}
