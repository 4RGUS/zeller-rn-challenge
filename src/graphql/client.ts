import {
  ApolloClient,
  ApolloLink,
  HttpLink,
  InMemoryCache,
} from "@apollo/client";
import awsconfig from "./aws-exports";
import { AuthOptions, createAuthLink } from "aws-appsync-auth-link";
import { createSubscriptionHandshakeLink } from "aws-appsync-subscription-link";

const auth: AuthOptions = {
  type: "API_KEY",
  apiKey: awsconfig.aws_appsync_apiKey,
};

const httpLink = new HttpLink({
  uri: awsconfig.aws_appsync_graphqlEndpoint,
});

const link = ApolloLink.from([
  createAuthLink({
    url: awsconfig.aws_appsync_graphqlEndpoint,
    region: awsconfig.aws_appsync_region,
    auth,
  }),
  createSubscriptionHandshakeLink(
    {
      url: awsconfig.aws_appsync_graphqlEndpoint,
      region: awsconfig.aws_appsync_region,
      auth,
    },
    httpLink
  ),
]);

const client = new ApolloClient({
  link,
  cache: new InMemoryCache(),
});

export default client;
