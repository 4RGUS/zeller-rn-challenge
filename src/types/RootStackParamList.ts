import { NativeStackNavigationProp } from "@react-navigation/native-stack";

export type RootStackParamList = {
  Home: undefined;
  UserList: undefined;
};

export type UserListScreenNavigationProp = NativeStackNavigationProp<
  RootStackParamList,
  "UserList"
>;
