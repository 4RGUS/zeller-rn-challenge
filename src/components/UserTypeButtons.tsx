import React from "react";
import { View, TouchableOpacity, Text, StyleSheet } from "react-native";
import { Roles } from "../graphql/types";
import { RadioCircle } from "./RadioCircle";

type UserTypeButtonsProps = {
  onPress: (role: Roles) => void;
  selectedRole: Roles;
};

export const UserTypeButtons = ({
  onPress,
  selectedRole,
}: UserTypeButtonsProps) => {
  return Object.values(Roles).map((role) => {
    const isSelected = selectedRole === role;
    const backgroundColor = isSelected ? "#E6F0FF" : "#FFF";
    return (
      <TouchableOpacity
        onPress={() => onPress(role)}
        key={role}
        style={[styles.container, { backgroundColor }]}
      >
        <RadioCircle selected={isSelected} />
        <Text>{role}</Text>
      </TouchableOpacity>
    );
  });
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    width: "100%",
    alignItems: "center",
    padding: 12,
    marginBottom: 8,
    borderRadius: 8,
    gap: 12,
  },
  button: {
    marginLeft: 12,
    width: "100%",
  },
});
