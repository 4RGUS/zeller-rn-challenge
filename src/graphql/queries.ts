import { gql } from "@apollo/client";

export const LIST_ZELLER_CUSTOMERS = gql`
  query ListZellerCustomers($filter: TableZellerCustomerFilterInput) {
    listZellerCustomers(filter: $filter) {
      items {
        id
        name
        email
        role
      }
      nextToken
    }
  }
`;

export const SEARCH_ZELLER_CUSTOMERS = gql`
  query SearchZellerCustomers(
    $nameFilter: TableStringFilterInput
    $limit: Int
    $nextToken: String
  ) {
    listZellerCustomers(
      filter: { name: $nameFilter }
      limit: $limit
      nextToken: $nextToken
    ) {
      items {
        id
        name
        email
        role
      }
      nextToken
    }
  }
`;
