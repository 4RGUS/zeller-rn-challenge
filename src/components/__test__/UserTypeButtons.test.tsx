import React from "react";
import { render } from "@testing-library/react-native";
import { UserTypeButtons } from "../UserTypeButtons";
import { Roles } from "../../graphql/types";

describe("UserTypeButtons component", () => {
  it("matches snapshot when Admin role is selected", () => {
    const { toJSON } = render(
      <UserTypeButtons onPress={jest.fn()} selectedRole={Roles.ADMIN} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("matches snapshot when Manager role is selected", () => {
    const { toJSON } = render(
      <UserTypeButtons onPress={jest.fn()} selectedRole={Roles.MANAGER} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
