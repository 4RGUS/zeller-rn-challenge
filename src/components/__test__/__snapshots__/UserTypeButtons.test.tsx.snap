// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTypeButtons component matches snapshot when Admin role is selected 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#E6F0FF",
        "borderRadius": 8,
        "flexDirection": "row",
        "gap": 12,
        "marginBottom": 8,
        "opacity": 1,
        "padding": 12,
        "width": "100%",
      }
    }
    testID="selected-role"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#4D94FF",
          "borderRadius": 9,
          "borderWidth": 2,
          "height": 18,
          "justifyContent": "center",
          "width": 18,
        }
      }
      testID="outer-circle"
    >
      <View
        style={
          {
            "backgroundColor": "#4D94FF",
            "borderRadius": 4,
            "height": 8,
            "width": 8,
          }
        }
        testID="inner-circle"
      />
    </View>
    <Text>
      ADMIN
    </Text>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 8,
        "flexDirection": "row",
        "gap": 12,
        "marginBottom": 8,
        "opacity": 1,
        "padding": 12,
        "width": "100%",
      }
    }
    testID="un-selected-role"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#4D94FF",
          "borderRadius": 9,
          "borderWidth": 2,
          "height": 18,
          "justifyContent": "center",
          "width": 18,
        }
      }
      testID="outer-circle"
    />
    <Text>
      MANAGER
    </Text>
  </View>,
]
`;

exports[`UserTypeButtons component matches snapshot when Manager role is selected 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 8,
        "flexDirection": "row",
        "gap": 12,
        "marginBottom": 8,
        "opacity": 1,
        "padding": 12,
        "width": "100%",
      }
    }
    testID="un-selected-role"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#4D94FF",
          "borderRadius": 9,
          "borderWidth": 2,
          "height": 18,
          "justifyContent": "center",
          "width": 18,
        }
      }
      testID="outer-circle"
    />
    <Text>
      ADMIN
    </Text>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#E6F0FF",
        "borderRadius": 8,
        "flexDirection": "row",
        "gap": 12,
        "marginBottom": 8,
        "opacity": 1,
        "padding": 12,
        "width": "100%",
      }
    }
    testID="selected-role"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#4D94FF",
          "borderRadius": 9,
          "borderWidth": 2,
          "height": 18,
          "justifyContent": "center",
          "width": 18,
        }
      }
      testID="outer-circle"
    >
      <View
        style={
          {
            "backgroundColor": "#4D94FF",
            "borderRadius": 4,
            "height": 8,
            "width": 8,
          }
        }
        testID="inner-circle"
      />
    </View>
    <Text>
      MANAGER
    </Text>
  </View>,
]
`;
